src/views/thread/container.js

```diff
@@ -146,6 +156,34 @@ class ThreadContainer extends React.Component<Props, State> {
     });
   }
 
+  handleScroll = e => {
+    e.persist();
+    if (!e || !e.target) return;
+
+    // whenever the user scrolls in the thread we determine if they've scrolled
+    // past the thread content section - once they've scroll passed it, we
+    // enable the `bannerIsVisible` state to slide the thread context banner
+    // in from the top of the screen
+    const scrollOffset = e.target.scrollTop;
+    this.setState({ scrollOffset }, () => {
```

Comment:
We shouldn't be putting the scroll offset (which changes every single frame you scroll) into the state, this is what's causing the issues. This isn't even necessary, just don't put it into the state, compare `scrollOffset` against `threadDetailHeight` and only set the state if it changes. That way you don't even need to debounce which is kinda annoying for this behavior

I'm taking a look at refactoring this!